validator/validators/performance/performance_validator.go

```diff
@@ -0,0 +1,239 @@
+	return &PerformanceValidator{
+		vConfig: vConfig,
+	}
+}
+
+func (s *PerformanceValidator) GenerateLoad() (err error) {
+	var (
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		dataType              = s.vConfig.GetDataType()
+		dataRate              = s.vConfig.GetDataRate()
+		receiver              = s.vConfig.GetPluginsConfig()
+	)
+	switch dataType {
+	case "logs":
+		err = common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, dataRate)
+	default:
+		err = common.StartSendingMetrics(receiver, agentCollectionPeriod, dataRate)
+	}
+
+	return err
+}
+
+func (s *PerformanceValidator) CheckData(startTime, endTime time.Time) error {
+	metrics, err := s.GetPerformanceMetrics(startTime, endTime)
+	if err != nil {
+		return err
+	}
+
+	perfInfo, err := s.CalculateMetricStatsAndPackMetrics(metrics.MetricDataResults)
+	if err != nil {
+		return err
+	}
+
+	err = s.SendPacketToDatabase(perfInfo)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *PerformanceValidator) Cleanup() error {
+	var (
+		dataType      = s.vConfig.GetDataType()
+		ec2InstanceId = awsservice.GetInstanceId()
+	)
+	switch dataType {
+	case "logs":
+		awsservice.DeleteLogGroup(ec2InstanceId)
+	}
+
+	return nil
+}
+
+func (s *PerformanceValidator) SendPacketToDatabase(perfInfo PerformanceInformation) error {
+	var (
+		dataType               = s.vConfig.GetDataType()
+		receiver               = s.vConfig.GetPluginsConfig()
+		commitHash, commitDate = s.vConfig.GetCommitInformation()
+		agentCollectionPeriod  = fmt.Sprint(s.vConfig.GetAgentCollectionPeriod().Seconds())
+		// The secondary global index that is used for checking if there are item has already been exist in the table
+		kCheckingAttribute = []string{"CommitDate", "UseCase"}
+		vCheckingAttribute = []string{fmt.Sprint(commitDate), receiver}
+	)
+
+	err := backoff.Retry(func() error {
+		existingPerfInfo, err := awsservice.GetItemInDatabase(DynamoDBDataBase, "UseCaseDate", kCheckingAttribute, vCheckingAttribute, perfInfo)
+		if err != nil {
+			return err
+		}
+
+		// Get the latest performance information from the database and update by merging the existing one
+		// and finally replace the packet in the database
+		maps.Copy(existingPerfInfo["Results"].(map[string]interface{}), perfInfo["Results"].(map[string]interface{}))
+
+		uniqueID := existingPerfInfo["UniqueID"].(string)
+		finalPerfInfo := packIntoPerformanceInformation(uniqueID, receiver, dataType, agentCollectionPeriod, commitHash, commitDate, existingPerfInfo["Results"])
+
+		err = awsservice.ReplaceItemInDatabase(DynamoDBDataBase, finalPerfInfo)
+
+		if err != nil {
+			return err
+		}
+		return nil
+	}, awsservice.StandardExponentialBackoff)
+
+	return err
+}
+func (s *PerformanceValidator) CalculateMetricStatsAndPackMetrics(metrics []types.MetricDataResult) (PerformanceInformation, error) {
```

Comment:
Calculate the metrics statistics of the MOM metrics (e.g procstat_cpu_usage) and return the average, min, max, stdev, p99. Similar to the [performance query utils](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/test/performancetest/performance_query_utils.go#L218-L263)